docs/code-search/types/search-jobs.mdx

```diff
@@ -83,10 +92,31 @@ To use Search Jobs, you need to:
 
 Search Jobs supports queries of `type:file` and it automatically appends this to the search query. Other result types (like `diff`, `commit`, `path`, and `repo`) will be ignored. However, there are some limitations on the supported query syntax. These include:
 
-- `OR`, `AND` operators
 - file predicates, such as `file:has.content`, `file:has.owner`, `file:has.contributor`, `file:contains.content`
 - `.*` regexp search
 - Multiple `rev` filters
 - Queries with `index: filter`
 
 <Callout type="note"> Sourcegraph already offers an [Exhaustive Search](/code-search/types/exhaustive) with the `count:all` operator. However, there are certain limitations when generating results within your codebase.</Callout>
+
+## FAQ
+
+### How can I run a search job against all branches and all repositories?
+
+You can use a combination of the `repo:` and `rev:` filter to search across all branches and repositories.
+For example, `r: rev:*/refs/heads/* foo` will search for `foo` across all branches and all repositories.
+See the [search syntax documentation](/code-search/queries#repository-revisions) for more details.
+
+### Is there a time limit for a search job?
+
+We break down a search job into smaller tasks and run them in parallel.
+Each task is limited to a single revision and a single repository.
+This allows us to run searches across large codebases.
+The chance of hitting timeouts is very low, but it's not impossible.
+If you do hit a timeout, you can try to break down your search query into smaller queries and run them separately.
```

Comment:
Should we encourage them reaching out to support here. Would love to debug when this happens / make it close to impossible.

Reply:
```suggestion
If you do hit a timeout, please reach out to support and we will help you troubleshoot the issue. As an immediate measure, you can break down your search query into smaller queries and run them separately.
```